### PR TITLE
Add test for SYN-391

### PR DESCRIPTION
**Pull request by erikjohnston**
https://matrix.org/jira/browse/SYN-391

### CODE REVIEW

File: tests/90jira/SYN-391.pl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+multi_test "Setting push rules twice 500s  SYN-391",
+    requires => [qw( register_new_user http_clients do_request_json_for )],
```
Comment (leonerd) [inline tests/90jira/SYN-391.pl]: blank line after here, between the named `multi_test` args

File: tests/90jira/SYN-391.pl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    do => sub {
+        my ( $register_new_user, $clients, $do_request_json_for ) = @_;
+        my $http = $clients->[0];
+        my $alice;
+
+        Future->needs_all(
```
Comment (leonerd) [inline tests/90jira/SYN-391.pl]: A `needs_all` of a single future is redundant, it can be removed.

File: tests/90jira/SYN-391.pl [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            $register_new_user->($http, "90jira-SYN-391_alice"),
```
Comment (leonerd) [inline tests/90jira/SYN-391.pl]: Any reason to be using `$register_new_user` here? If the requirements on the user aren't anything overly special, just use a `local_user_fixture()` like most of the other tests, to get a new user automatically provisioned.

File: tests/90jira/SYN-391.pl [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        )->then( sub {
+            ($alice) = @_;
```
Comment (leonerd) [inline tests/90jira/SYN-391.pl]: Wants whitespace inside parens `( $vars, here.... )`
Wants blank line after the `@_`-unpacking line.

**Comment by erikjohnston** [comment]
(This PR by @negativemjark)

**Comment by NegativeMjark** [comment]
@leonerd PTAL

File: tests/61push/02add_rules.pl [hunk 5]
```diff
@@ -317,3 +317,34 @@ sub check_add_push_rule
...
+test "Adding the same push rule twice is idempotent",
+   requires => [ local_user_fixture() ],
+
+   bug => "SYN-391",
+
+   do => sub {
+      my ( $user ) = @_;
+
+      matrix_add_push_rule( $user, "global", "sender", '@bob:example.com', {
+         "actions" => [ "notify" ]
```
Comment (leonerd) [inline tests/61push/02add_rules.pl]: The quote marks here are unnecessary; as it's a bareword on the LHS of the fat arrow, it'll quote nicely anyway

File: tests/61push/02add_rules.pl [hunk 6]
```diff
@@ -317,3 +317,34 @@ sub check_add_push_rule
...
+         matrix_get_push_rules( $user );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( my $global = $body->{global}, qw( sender ) );
+         assert_json_list( my $sender = $global->{sender} );
+
+         assert_eq( $sender->[0]{rule_id}, '@bob:example.com' );
+
+         @{ $sender } == 1 or die "Expected only one rule";
```
Comment (leonerd) [inline tests/61push/02add_rules.pl]: The braces are also unnecessary here; `@$sender` is clear enough. As per style-guide this latter form is preferred.

**Comment by leonerd** [comment]
LGTM besides a couple of very minor style issues

**Pull request merged**